hooks/nvidia-bootstrap/image/run.sh

```diff
@@ -26,43 +26,80 @@ CACHE_DIR_CONTAINER="${ROOTFS_DIR}${CACHE_DIR_HOST}"
 # AWS Instance Types to Nvidia Card Mapping (cut and pasted from AWS docs)
 # Load the correct driver for the correct instance type
 #   Instances  Product Type  Product Series  Product
-#   G2         GRID          GRID Series     GRID K520   <-- I think they meant G3
+#   G2         GRID          GRID Series     GRID K520 (deprecated)
+#   G3         Tesla         M-Series        M-60
+#   G3S        Tesla         M-Series        M-60
+#   G4         Tesla         T-Series        T-4
 #   P2         Tesla         K-Series        K-80
 #   P3         Tesla         V-Series        V100
 # Both P2 and P3 are set for Cuda Toolkit 9.1
 # http://www.nvidia.com/Download/index.aspx
 declare -A class_to_driver_file
-class_to_driver_file=( \
-    ["g3"]="http://us.download.nvidia.com/XFree86/Linux-x86_64/367.124/NVIDIA-Linux-x86_64-367.124.run" \
-    ["g3s"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
-    ["p2"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
-    ["p3"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
-)
 declare -A class_to_driver_checksum
-class_to_driver_checksum=( \
-    ["g3"]="77f37939efeea4b6505842bed50445971992e303" \
-    ["g3s"]="57569ecb6f6d839ecc77fa10a2c573cc069990cc" \
-    ["p2"]="57569ecb6f6d839ecc77fa10a2c573cc069990cc" \
-    ["p3"]="57569ecb6f6d839ecc77fa10a2c573cc069990cc" \
-)
-
-# CUDA Files that need to be installed ~1.4GB
-#   First one is main installation
-#   Subsequent files are patches which need to be applied in order
-#   Order in the arrays below matters
-# https://developer.nvidia.com/cuda-downloads
-cuda_files=( \
-  "https://developer.nvidia.com/compute/cuda/9.1/Prod/local_installers/cuda_9.1.85_387.26_linux" \
-  "https://developer.nvidia.com/compute/cuda/9.1/Prod/patches/1/cuda_9.1.85.1_linux" \
-  "https://developer.nvidia.com/compute/cuda/9.1/Prod/patches/2/cuda_9.1.85.2_linux" \
-  "https://developer.nvidia.com/compute/cuda/9.1/Prod/patches/3/cuda_9.1.85.3_linux" \
-)
-cuda_files_checksums=( \
-  "1540658f4fe657dddd8b0899555b7468727d4aa8" \
-  "7ec6970ecd81163b0d02ef30d35599e7fd6e97d8" \
-  "cfa3b029b58fc117d8ce510a70efc848924dd565" \
-  "6269a2c5784b08997edb97ea0020fb4e6c8769ed" \
-)
+case $CUDA_VERSION in
+    9.1)
+        class_to_driver_file=( \
+            ["g3"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
+            ["g3s"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
+            ["g4dn"]="http://us.download.nvidia.com/tesla/418.87/NVIDIA-Linux-x86_64-390.46.00.run" \
```

Comment:
1. This driver returns a 404
2. Is it intentional that g4dn uses another driver?

```suggestion
            ["g4dn"]="http://us.download.nvidia.com/tesla/390.46/NVIDIA-Linux-x86_64-390.46.run" \
```

Reply:
Just a bad copypasta